shopBuddy/shopBuddy/APIManager.m

```diff
@@ -24,23 +25,26 @@ - (instancetype) init {
     if(self=[super init])
     {
         manager = [[AFHTTPSessionManager alloc] initWithBaseURL:[NSURL URLWithString:baseURLString]];
+        NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"]];
+        key = [dict objectForKey: @"api_key"];
     }
     return self;
 }
 
-- (void)getItemWithBarcode:(NSString *)barcode completion:(void(^)(NSDictionary *itemDetails, NSError *error))completion
+- (void)getItemWithBarcode:(NSString *)barcode completion:(void(^)(Item *item, NSError *error))completion
 {
 
     NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"]];
     NSString *key = [dict objectForKey: @"api_key"];
     //NSString *barcode_call =@"v3/products?barcode=3614272049529&formatted=y&key=";
-    NSString *path = [NSString stringWithFormat:@"v3/products?barcode=%@&formatted=y&key=%@", barcode, key];
+    NSString *const path = [NSString stringWithFormat:@"v3/products?barcode=%@&formatted=y&key=%@", barcode, key];
 
     [manager GET:path parameters:nil headers: nil progress:nil success:^(NSURLSessionTask *task, NSDictionary *responseObject)
      {
          // Success
          NSLog(@"Success: %@", responseObject);
-        completion(responseObject, nil);
+        Item *item = [[Item alloc] initWithDictionary:responseObject[@"products"][0]];
```

Comment:
Usage of `const` in all object declaration that are not intended to change provide a way to ensure the value is changed by accident since the compiler will show an error if it happens.